### PR TITLE
Add Generics 

### PR DESCRIPTION
**Pull request by AlexSandraNmk**
- Changed methods to be able to work with generics, not only ints.

### CODE REVIEW

File: SortingAlgorithms/Interfaces/ISortingAlgorithm.cs [hunk 1]
```diff
@@ -8,6 +8,6 @@ namespace SortingAlgorithms.Interfaces
 {
     interface ISortingAlgorithm
     {
-        void Sort(int[] array);
+        void Sort<T>(T[] array) where T : IComparable;
```
Comment (aleksey-efremov) [inline SortingAlgorithms/Interfaces/ISortingAlgorithm.cs:11]: Please, add xml comments to all methods. You can use <inheritdoc/> for inheritors.
For example, here, method Sort should be filled xml comments and BinarySearch and LinearSearch can be marked as <inheritdoc/>.

File: SortingAlgorithms/Helpers/SortedChecker.cs [hunk 2]
```diff
@@ -8,11 +8,11 @@ namespace SortingAlgorithms.Helpers
 {
     public static class SortedChecker
```
Comment (aleksey-efremov) [inline SortingAlgorithms/Helpers/SortedChecker.cs:9]: Please, avoid static classes. 
For me it better to call ArrayValidator.

File: SortingAlgorithms/Algorithms/Sorting/MergeSort.cs [hunk 3]
```diff
@@ -9,12 +9,12 @@ namespace SortingAlgorithms.Algorithms.Sorting
 {
     public class MergeSort : ISortingAlgorithm
```
Comment (aleksey-efremov) [inline SortingAlgorithms/Algorithms/Sorting/MergeSort.cs:10]: Class could be marked as sealed
The same for others files

File: SortingAlgorithms/Algorithms/Searching/BinarySearch.cs [hunk 4]
```diff
@@ -11,44 +11,44 @@ namespace SortingAlgorithms.Algorithms.Searching
 {
     public class BinarySearch : ISearchingAlgorithm
     {
-        public void Search(int[] array, int number)
+        public void Search<T>(T[] array, T item) where T : IComparable
         {
             if (SortedChecker.IsSorted(array) is false)
             {
                 ISortingAlgorithm sortingAlgorithm = new MergeSort();
```
Comment (aleksey-efremov) [inline SortingAlgorithms/Algorithms/Searching/BinarySearch.cs:18]: Better to use it as property and set from constructor
The same for SortedChecker, make it dynamic and set from ctor

**Review by aleksey-efremov — APPROVED** [review]

File: SortingAlgorithms/Helpers/SortedChecker.cs (hunk 2 above)
Comment (AlexSandraNmk) [reply SortingAlgorithms/Helpers/SortedChecker.cs]: done

File: SortingAlgorithms/Algorithms/Searching/BinarySearch.cs [hunk 5]
```diff
@@ -9,46 +9,60 @@
 
 namespace SortingAlgorithms.Algorithms.Searching
 {
-    public class BinarySearch : ISearchingAlgorithm
+    public sealed class BinarySearch : ISearchingAlgorithm 
     {
-        public void Search(int[] array, int number)
+        private readonly ISortingAlgorithm _sortingAlgorithm;
+        private readonly dynamic _arrayValidator;
```
Comment (aleksey-efremov) [inline SortingAlgorithms/Algorithms/Searching/BinarySearch.cs:15]: Dynamic is a bad practice, for me.
Create interface IArrayValidator and use the same way as ISortingAlgorithm

**Review by aleksey-efremov — COMMENTED** [review]
Looks good!

**Pull request merged**